specs/governance/gov-token.md

```diff
@@ -5,93 +5,68 @@
+`GovernanceToken` MUST have a `mint(address,uint256)` function with external visibility that allows the `owner()` address
 to mint an arbitrary number of new tokens to a specific address. This function MUST only be called by the contract
-owner, the `MintManager`, as enforced by the `onlyOwner` modifier inherited from the `Ownable` contract. When tokens
-are minted, the voting power of the recipient address MUST be updated accordingly in the `GovernanceDelegation` contract
-via the `afterTokenTransfer` hook. The total token supply is capped to `2^208 - 1` to prevent overflow risks in the voting
-system. If the total supply exceeds this limit, `_mint(address,uint256)`, as inherited from `ERC20Votes`, MUST revert.
+owner, the `MintManager`, as enforced by the `onlyOwner` modifier inherited from the `Ownable` contract.
 
 ### Token Burning
 
 The contract MUST allow token holders to burn their own tokens using the inherited `burn(uint256)` or
-`burnFrom(address,uint256)` functions inherited from `ERC20Burnable`. When tokens are burned, the total supply and the
-holder's voting power MUST be reduced accordingly in the `GovernanceDelegation` contract via the `afterTokenTransfer` hook.
+`burnFrom(address,uint256)` functions inherited from `ERC20Burnable`. Burn functions MUST NOT allow a user's balance to
+underflow and attempts to reduce balance below zero MUST revert.
 
 ### Voting Power
 
-Each token corresponds to one unit of voting power.
-By default, token balance does not account for voting power. To have their voting power counted, token holders MUST delegate
-their voting power to an address (can be their own address).
-The contract MUST offer public accessors for querying voting power, as outlined below.
-
-#### Queries
-
-- The `getVotes(address)(uint256)` function MUST retrieve the current voting power of an address from the
-  `GovernanceDelegation` contract.
-- The `getPastVotes(address,uint256)(uint256)` function MUST allow querying the voting power of an address at a specific
-  block number in the past from the `GovernanceDelegation` contract.
-- The `getPastTotalSupply(uint256)(uint256)` function MUST return the total voting power at a specific block number in
-  the past from the `GovernanceDelegation` contract.
+Each token corresponds to one unit of voting power. Token holders who wish to utilize their tokens for voting MUST delegate
+their voting power to an address (can be their own address). An active delegation MUST NOT prevent a user from transferring
+tokens. The `GovernanceToken` contract MUST offer public accessor functions for querying voting power, as outlined below.
+
+#### Public Query Functions
+
+- `checkpoints(address,uint32) returns (Checkpoint)`
+  - MUST retrieve the n-th Checkpoint for a given address.
+- `numCheckpoints(address) returns (uint32)`
+  - MUST retrieve the total number of Checkpoint objects for a given address.
```

Comment:
we also have these functions in the experiemental version (they fetch from govdelegation if address has been migrated)